gobblin-core/src/main/java/gobblin/source/DatePartitionedAvroFileSource.java

```diff
@@ -125,15 +150,25 @@
   private int fileCount;
   private TableType tableType;
   private Path sourceDir;
-
+  private String sourcePartitionPrefix;
+  private String sourcePartitionSuffix;
+  private DateTimeFormatter partitionPatternFormatter;
+  private DurationFieldType incrementalUnit;
+  
   /**
    * Gobblin calls the {@link Source#getWorkunits(SourceState)} method after creating a {@link Source} object with a
    * blank constructor, so any custom initialization of the object needs to be done here.
    */
   private void init(SourceState state) {
     DateTimeZone.setDefault(DateTimeZone
         .forID(state.getProp(ConfigurationKeys.SOURCE_TIMEZONE, ConfigurationKeys.DEFAULT_SOURCE_TIMEZONE)));
-
+    
+    initDatePartitionFromPattern(state);
```

Comment:
Have one method called `initDatePartition(State state)` that sets the datePattern either from granularity or pattern key.